spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/batch/SimpleJobService.java

```diff
@@ -263,6 +238,18 @@ public int stopAll() {
 		return result.size();
 	}
 
+	/**
+	 * Signal the {@link JobExecution} to stop. Iterates through the associated
+	 * {@link StepExecution}s, calling {@link StepExecution#setTerminateOnly()}.
+	 *
+	 */
+	public void stopJobExecution(JobExecution jobExecution) {
+		for (StepExecution stepExecution : jobExecution.getStepExecutions()) {
+			stepExecution.setTerminateOnly();
+		}
```

Comment:
```suggestion
		jobExecution.getStepExecutions().forEach(StepExecution::setTerminateOnly);
```
This can be simplified a bit w/ the above. Wdyt?